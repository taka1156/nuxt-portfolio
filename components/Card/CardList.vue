<template>
  <div class="CardLayout">
    <div class="cardlist">
      <div v-for="(cardInfo, index) in cards" :key="index" class="card">
        <card-layout :card-info="cardInfo" />
      </div>
    </div>
  </div>
</template>

<script>
import CardLayout from './CardLayout';

export default {
  name: 'CardLayout',
  components: {
    'card-layout': CardLayout
  },
  props: {
    cards: {
      type: Array,
      default: () => []
    }
  }
};
</script>

<style scoped>
.cardlist {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card {
  width: 27rem;
  cursor: pointer;
  margin: 2em auto;
  border: solid 0.6px lightgray;
  border-radius: 1%;
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}
</style>
