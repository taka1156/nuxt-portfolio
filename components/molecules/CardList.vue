<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-item
        v-for="(cardInfo, index) in cards"
        :key="index"
        :card-info="cardInfo"
        class="card-list__card"
      ></card-item>
    </div>
  </div>
</template>

<script>
import CardListItem from '../molecules/CardListItem';

export default {
  name: 'CardList',
  components: {
    'card-item': CardListItem,
  },
  props: {
    cards: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 27em;
  margin: 2.6em auto;
}
</style>
