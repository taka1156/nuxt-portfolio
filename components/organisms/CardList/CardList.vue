<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-list-item
        v-for="card in cards"
        :key="card.name"
        :card="card"
        class="card-list__card"
      />
    </div>
  </div>
</template>

<script>
import CardListItem from '../../molecules/CardListItem/CardListItem';

export default {
  name: 'CardList',
  components: {
    CardListItem
  },
  props: {
    /**
     * カードリスト
     */
    cards: {
      type: Array,
      required: true
    }
  }
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 25em;
  margin: 0 auto 2em;
}
</style>
