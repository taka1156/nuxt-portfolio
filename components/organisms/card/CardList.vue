<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-list-item
        v-for="(card, index) in cards"
        :key="index"
        :card="card"
        class="card-list__card"
        @jump="jump"
      />
    </div>
  </div>
</template>

<script>
import CardListItem from '../../molecules/card/CardListItem';

export default {
  name: 'CardList',
  components: {
    'card-list-item': CardListItem,
  },
  props: {
    /**
     * カードリスト
     */
    cards: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    /**
     * 制作物関連サイトに移動
     * (githubやwebサイトのリンク先)
     */
    jump(card = null) {
      if (card == null || card.link == null) {
        return;
      }
      location.assign(card.link);
    },
  },
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 28em;
  margin: 0 auto 2em;
}
</style>
