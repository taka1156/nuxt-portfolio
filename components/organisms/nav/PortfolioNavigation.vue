<template>
  <div>
    <nav>
      <nav-bar :logo-text="logoText" :is-open="isOpen" @change-state="changeState" />
      <nav-list :is-open="isOpen" :routes="routes" @change-state="changeState" />
    </nav>
  </div>
</template>

<script>
import NavBar from '../../molecules/nav/NavBar';
import NavList from '../../molecules/nav/NavList';

export default {
  name: 'BaseNav',
  components: {
    'nav-bar': NavBar,
    'nav-list': NavList,
  },
  props: {
    logoText: {
      type: String,
      default: 'Please Setting Title',
      required: true,
    },
    routes: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  data() {
    return {
      isOpen: false,
    };
  },
  methods: {
    changeState() {
      // ナビゲーションバーの開閉管理
      this.isOpen = !this.isOpen;
    },
  },
};
</script>
