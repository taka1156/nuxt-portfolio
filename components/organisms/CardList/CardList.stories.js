import CardList from './CardList.vue';
import { dummyCards, dummyCardsLink } from '@/@/__testdata__/testdata.js';

export default {
  title: 'Organisms/CardList',
  component: CardList
};

const Template1 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardList },
  template: '<card-list v-bind="$props" />'
});

const Template2 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardList },
  template: '<card-list v-bind="$props" />'
});

export const Default = Template1.bind({});
Default.args = { cards: dummyCards };

export const Link = Template2.bind({});
Link.args = { cards: dummyCardsLink };
