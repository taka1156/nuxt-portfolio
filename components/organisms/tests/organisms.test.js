import { mount, RouterLinkStub } from '@vue/test-utils';
import PortfolioNavigation from '../PortfolioNavigation.vue';
import CardList from '../CardList.vue';

// BaseNav
describe('BaseNav', () => {
  const dummyLogo = 'ダミー';
  const dummyRoutes = [
    { name: 'Top', to: '/home', img: 'http://placehold.jp/150x150.png' },
  ];
  const baseNav = mount(PortfolioNavigation, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: {
      logoText: dummyLogo,
      routes: dummyRoutes,
    },
  });

  it('NavBar初期値: logoText, routes', () => {
    // logoText
    expect(baseNav.vm.$options.props.logoText.required).toBe(true);
    expect(baseNav.vm.logoText).toBe(dummyLogo);
    // routes
    expect(baseNav.vm.$options.props.routes.required).toBe(true);
    console.log(baseNav.vm.$options.props.routes.default());
    expect(baseNav.vm.routes).toBe(dummyRoutes);
  });

  it('閉じた時のスナップショット', () => {
    expect(baseNav.html()).toMatchSnapshot();
  });

  it('changeStateのテスト', () => {
    expect(baseNav.vm.$data.isOpen).toBe(false);
    baseNav.vm.changeState();
    expect(baseNav.vm.$data.isOpen).toBe(true);
  });

  it('開いた時のスナップショット', () => {
    expect(baseNav.html()).toMatchSnapshot();
  });
});

// BaseCard
describe('BaseCads', () => {
  const cardList = propsData => {
    return mount(CardList, {
      propsData: {
        ...propsData,
      },
    });
  };

  const dummyCards = [
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
    },
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
    },
  ];

  const dummyCardsLink = [
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
      link: 'https://taka1156.site',
    },
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
      link: 'https://github.com/taka1156/nuxt-portoflio',
    },
  ];

  it('CardListの初期値(linkなし): dummyCards', () => {
    const wrapper = cardList({ cards: dummyCards });
    console.log(wrapper.vm.$options.props.cards.default());
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(wrapper.vm.cards).toBe(dummyCards);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('CardListの初期値(linkあり): dummyCardsLink', () => {
    const wrapper = cardList({ cards: dummyCardsLink });
    const btnTag = wrapper.findAll('button');
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(btnTag.at(0).text()).toBe('WebSite');
    expect(btnTag.at(1).text()).toBe('GitHub');
    expect(wrapper.vm.cards).toBe(dummyCardsLink);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
