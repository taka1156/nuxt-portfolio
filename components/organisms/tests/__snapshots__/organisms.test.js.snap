// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCads CardListの初期値(linkあり): dummyCardsLink 1`] = `
"<div>
  <div>
    <div class=\\"card-list\\">
      <div class=\\"card-list__card\\">
        <article class=\\"card__item\\">
          <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card__img\\">
            <figcaption>
              <h2 class=\\"card__title\\">ダミータイトル</h2>
              <p class=\\"card__text\\">詳細</p>
            </figcaption>
          </figure> <button class=\\"btn\\">
            WebSite
          </button>
        </article>
      </div>
      <div class=\\"card-list__card\\">
        <article class=\\"card__item\\">
          <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card__img\\">
            <figcaption>
              <h2 class=\\"card__title\\">ダミータイトル</h2>
              <p class=\\"card__text\\">詳細</p>
            </figcaption>
          </figure> <button class=\\"btn\\">
            GitHub
          </button>
        </article>
      </div>
    </div>
  </div>
</div>"
`;

exports[`BaseCads CardListの初期値(linkなし): dummyCards 1`] = `
"<div>
  <div>
    <div class=\\"card-list\\">
      <div class=\\"card-list__card\\">
        <article class=\\"card__item\\">
          <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card__img\\">
            <figcaption>
              <h2 class=\\"card__title\\">ダミータイトル</h2>
              <p class=\\"card__text\\">詳細</p>
            </figcaption>
          </figure>
          <!---->
        </article>
      </div>
      <div class=\\"card-list__card\\">
        <article class=\\"card__item\\">
          <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card__img\\">
            <figcaption>
              <h2 class=\\"card__title\\">ダミータイトル</h2>
              <p class=\\"card__text\\">詳細</p>
            </figcaption>
          </figure>
          <!---->
        </article>
      </div>
    </div>
  </div>
</div>"
`;

exports[`BaseNav 閉じた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav__logo\\">ダミー</a>
          <div>
            <div class=\\"border__group\\"><span class=\\"border__top border__top--close\\"></span> <span class=\\"border__middle\\"></span> <span class=\\"border__bottom border__bottom--close\\"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"display: none;\\">
          <ul>
            <div>
              <li class=\\"nav__item\\"><a class=\\"nav__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav__icon\\">
                  <div class=\\"nav__text\\">Top</div>
                </a></li>
            </div>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;

exports[`BaseNav 開いた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav__logo\\">ダミー</a>
          <div>
            <div class=\\"border__group\\"><span class=\\"border__top border__top--open\\"></span> <span class=\\"border__middle border__middle--fade\\"></span> <span class=\\"border__bottom border__bottom--open\\"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"\\">
          <ul>
            <div>
              <li class=\\"nav__item\\"><a class=\\"nav__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav__icon\\">
                  <div class=\\"nav__text\\">Top</div>
                </a></li>
            </div>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;
